travis_docs_builder.py

```diff
@@ -167,13 +167,48 @@ def setup_GitHub_push(repo):
         'https://{token}@github.com/{repo}.git'.format(token=token.decode('utf-8'), repo=repo)])
     print("Fetching token remote")
     run(['git', 'fetch', 'origin_token'])
+    #create gh-pages empty branch with .nojekyll if it doesn't already exist
+    create_gh_pages()
     print("Checking out gh-pages")
-    # TODO: Create the remote gh-pages if it doesn't exist
     run(['git', 'checkout', '-b', 'gh-pages', '--track', 'origin_token/gh-pages'])
     print("Done")
 
     return True
 
+def gh_pages_exists():
+    """
+    Check if there is a remote gh-pages branch.
+    This isn't completely robust.  If there are multiple remotes and you have a gh-pages
+    branch on the non-default remote, this won't see it.
+    """
+    remote_name = 'origin_token'
+    branch_names = subprocess.check_output(['git', 'branch', '-r']).decode('utf-8').split()
+
+    return '{}/gh-pages'.format(remote_name) in branch_names
+
+def create_gh_pages():
+    """
+    If there is no remote gh-pages branch, create one
+
+    Return True if gh-pages was created, False if not
+    """
+    if not gh_pages_exists():
+        print("Creating gh-pages branch")
+        run(['git', 'symbolic-ref', 'HEAD', 'refs/heads/gh-pages'])
```

Comment:
Supposedly you can do `git checkout --orphan` as well http://stackoverflow.com/a/13969482/161801